daemon/api_test.go

```diff
@@ -134,8 +130,7 @@ oem: {store: {id: %q}}
 }
 
 func (s *apiSuite) TestPackageInfoOneIntegration(c *check.C) {
-	d := New()
-	d.addRoutes()
+	newTestDaemon()
```

Comment:
It's not any worse, but I don't see much improvement in that sense either. The problem is that New is not really New. The call to addRoutes overwrites global state and makes it active in a place that only once such instance can be active, and it also does not de-initializes it, meaning in can even leak behavior into other tests.

A better abstraction we can evolve towards might be along the lines of a self-contained daemon that is started and stopped with the test (or used via the handler API), and that holds all state internally. To use it, either we poke it via its port, or if using via the handler interface we simply let it die off, but then without any global reference to it.